cppcache/src/LocalRegion.cpp

```diff
@@ -346,15 +348,16 @@ void LocalRegion::localPut(const CacheableKeyPtr& key,
   GfErrTypeToException("Region::localPut", err);
 }
 
-void LocalRegion::putAll(const HashMapOfCacheable& map, uint32_t timeout,
+void LocalRegion::putAll(const HashMapOfCacheable& map,
+                         std::chrono::milliseconds timeout,
                          const SerializablePtr& aCallbackArgument) {
-  if ((timeout * 1000) >= 0x7fffffff) {
-    throw IllegalArgumentException(
-        "Region::putAll: timeout parameter "
-        "greater than maximum allowed (2^31/1000 i.e 2147483).");
-  }
-  int64_t sampleStartNanos = startStatOpTime();
-  GfErrType err = putAllNoThrow(map, timeout, aCallbackArgument);
+  //  if ((timeout * 1000) >= 0x7fffffff) {
```

Comment:
TODO?

Reply:
Shoot, yes!

Reply:
@dgkimura I had really hoped that redefining the parameter as `std::chrono::duration<int32_t, std::milli> timeout` would have solved it compile time but strangely it doesn't. I was able to compile a call at passed in64_t max seconds, which is WAY over the limit. I feel like this is compiler bug but we will just have to check at runtime now.

Reply:
@dgkimura Fixed in https://github.com/apache/geode-native/pull/144/commits/91a47a9f898921932ec037704bde43edf3ba04b4